src/senaite/core/exportimport/instruments/importer.py

```diff
@@ -0,0 +1,1049 @@
+            RenderInReport=False,
+        )
+        attachment.reindexObject()
+
+        logger.info(_(
+            "Attached file '%{filename}' to worksheet ${worksheet}",
+            mapping={
+                "filename": filename,
+                "worksheet": ws.getId(),
+            }))
+
+        return attachment
+
+    def create_mime_attachmenttype(self):
+        """Create (or get) an attachment filetype
+        """
+        file_type = self.parser.getAttachmentFileType()
+        obj = self.get_attachment_type_by_title(file_type)
+        if not obj:
+            obj = api.create(self.attachment_types, "AttachmentType")
+            obj.edit(title=file_type,
+                     description="Auto generated")
+        return obj
+
+    def attach_attachment(self, analysis, attachment):
+        """Attach a file or a given set of files to an analysis
+
+        :param analysis: analysis where the files are to be attached
+        :param attachment: files to be attached. This can be either a
+        single file or a list of files
+        :return: None
+        """
+        if not attachment:
+            return
+        if isinstance(attachment, list):
+            for attach in attachment:
+                self.attach_attachment(analysis, attach)
+            return
+        # current attachments
+        an_atts = analysis.getAttachment()
+        atts_filenames = [att.getAttachmentFile().filename for att in an_atts]
+        filename = attachment.getAttachmentFile().filename
+
+        if filename not in atts_filenames:
+            an_atts.append(attachment)
+            logger.info(
+                _("Attaching '${attachment}' to Analysis '${analysis}'",
+                  mapping={
+                      "attachment": filename,
+                      "analysis": analysis.getKeyword(),
+                  }))
+            analysis.setAttachment([att.UID() for att in an_atts])
+            analysis.reindexObject()
+        else:
+            self.log(_("Attachment '${attachment}' was already linked "
+                       "to analysis ${analysis}",
+                       mapping={
+                           "attachment": filename,
+                           "analysis": analysis.getKeyword(),
+                       }))
+
+    def get_attachment_filenames(self, ws):
+        """Returns all attachment filenames in the given worksheet
+        """
+        fn_attachments = {}
+        for att in ws.objectValues("Attachment"):
+            fn = att.getAttachmentFile().filename
+            if fn not in fn_attachments:
+                fn_attachments[fn] = []
+            fn_attachments[fn].append(att)
+        return fn_attachments
+
+    def is_analysis_allowed(self, analysis):
+        """Filter analyses that match the import criteria
+        """
+        portal_type = api.get_portal_type(analysis)
+        if portal_type != "Analysis":
+            # Reference Analysis / Duplicate Analysis
+            return True
```

Comment:
I'd say this is used to allow the update of `ReferenceAnalysis` objects even when they were already verified, so if the new result is within ranges (while previous not), the instrument would automatically be considered as valid. This probably deserves a thorough discussion in another PR, but for now I would at least keep "DuplicateAnalyses" out of the equation. So I would replace this condition as follows:

```python
if IReferenceAnalysis.providedBy(analysis):
    return True
```

Reply:
Refactored in d0ec1c2ed